src/autoComplete/index.tsx

```diff
@@ -168,19 +192,24 @@ export default class AutoComplete extends Component<AutoCompleteProps, AutoCompl
   onInputChange = ({ value }: { value: query }) => {
     this.currentValue = value
     if (this.hasMinCharsForSearch()) {
-      this.setState({ noResults: false, query: value }, this.searchForItems)
+      this.setState({
+        textfieldValue: String(value),
```

Comment:
It's a good idea to separate the textfieldValue from the query as they have different concerns. Can you check however if we can remove `this.currentValue` it seems to be just a copy of the query.

Reply:
Removed (I paramaterized the hasMinCharsForSearch function instead)